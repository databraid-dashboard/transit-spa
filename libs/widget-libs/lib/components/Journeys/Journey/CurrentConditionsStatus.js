var _jsxFileName = '../../src/components/Journeys/Journey/CurrentConditionsStatus.jsx',
    _this = this;

import React from 'react';
import PropTypes from 'prop-types';

import { Icon, List, Popup } from 'semantic-ui-react';

var style = {
  borderRadius: 0,
  opacity: 0.7,
  padding: '2em'
};

var CurrentConditionsStatus = function CurrentConditionsStatus(_ref) {
  var conditionStatus = _ref.conditionStatus;
  return React.createElement(
    List.Item,
    {
      __source: {
        fileName: _jsxFileName,
        lineNumber: 13
      },
      __self: _this
    },
    React.createElement(
      List.Content,
      {
        __source: {
          fileName: _jsxFileName,
          lineNumber: 14
        },
        __self: _this
      },
      React.createElement(
        List.Header,
        {
          __source: {
            fileName: _jsxFileName,
            lineNumber: 15
          },
          __self: _this
        },
        'status:'
      ),
      React.createElement(Popup, {
        trigger: React.createElement(Icon, {
          name: conditionStatus === 'on-time' ? 'checkmark' : 'exclamation triangle',
          size: 'large',
          __source: {
            fileName: _jsxFileName,
            lineNumber: 18
          },
          __self: _this
        }),
        content: conditionStatus,
        style: style,
        offset: 50,
        position: 'right center',
        inverted: true,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 16
        },
        __self: _this
      })
    )
  );
};

CurrentConditionsStatus.propTypes = {
  conditionStatus: PropTypes.string.isRequired
};

CurrentConditionsStatus.defaultProps = {
  conditionStatus: ''
};

export default CurrentConditionsStatus;