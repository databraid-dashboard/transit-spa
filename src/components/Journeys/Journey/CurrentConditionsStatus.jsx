import React from 'react';
import PropTypes from 'prop-types';

import { Popup, Label } from 'semantic-ui-react';

const labelStyle = {
  marginTop: '0.5rem',
  marginBottom: '0rem',
};

const popupStyle = {
  borderRadius: 0,
  opacity: 0.7,
  padding: '2em',
};

const CurrentConditionsStatus = ({ conditionStatus }) => {
  const color = conditionStatus === 'on-time' ? 'green' : 'red';
  const status = conditionStatus === 'on-time' ? 'on-time' : 'delayed';

  return (
    <Popup
      trigger={
        <Label style={labelStyle} color={conditionStatus === null ? 'grey' : color}>
          {conditionStatus === null ? 'unknown' : status}
        </Label>
      }
      content={conditionStatus}
      style={popupStyle}
      offset={50}
      position="right center"
      inverted
    />
  );
};

CurrentConditionsStatus.propTypes = {
  conditionStatus: PropTypes.string.isRequired,
};

CurrentConditionsStatus.defaultProps = {
  conditionStatus: 'unknown',
};

export default CurrentConditionsStatus;
