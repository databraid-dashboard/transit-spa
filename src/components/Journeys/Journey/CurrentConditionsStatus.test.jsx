import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import CurrentConditionsStatus from './CurrentConditionsStatus';

describe('CurrentConditionsStatus', () => {
  it('renders when given an on-time status', () => {
    const component = shallow(<CurrentConditionsStatus conditionStatus={'on-time'} />);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('renders when given a delayed status', () => {
    const component = shallow(
      <CurrentConditionsStatus conditionStatus={'delayed for some time due to reasons'} />,
    );
    expect(toJson(component)).toMatchSnapshot();
  });
});
