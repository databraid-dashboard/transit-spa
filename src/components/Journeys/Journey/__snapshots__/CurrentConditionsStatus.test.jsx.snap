// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentConditionsStatus renders when given a delayed status 1`] = `
<Popup
  content="delayed for some time due to reasons"
  inverted={true}
  offset={50}
  on="hover"
  position="right center"
  style={
    Object {
      "borderRadius": 0,
      "opacity": 0.7,
      "padding": "2em",
    }
  }
  trigger={
    <Label
      color="red"
      style={
        Object {
          "marginBottom": "0rem",
          "marginTop": "0.5rem",
        }
      }
    >
      delayed
    </Label>
  }
/>
`;

exports[`CurrentConditionsStatus renders when given an on-time status 1`] = `
<Popup
  content="on-time"
  inverted={true}
  offset={50}
  on="hover"
  position="right center"
  style={
    Object {
      "borderRadius": 0,
      "opacity": 0.7,
      "padding": "2em",
    }
  }
  trigger={
    <Label
      color="green"
      style={
        Object {
          "marginBottom": "0rem",
          "marginTop": "0.5rem",
        }
      }
    >
      on-time
    </Label>
  }
/>
`;
